File: cardView/app/src/main/java/com/example/cardview/MainActivity.java
package com.example.cardview;

import androidx.appcompat.app.AppCompatActivity;
import androidx.cardview.widget.CardView;

import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        CardView cardView;

        cardView = findViewById(R.id.cardding);
        cardView.setCardElevation(15.0f);
        cardView.setUseCompatPadding(true);

    }
}

File: webView/app/src/main/java/com/example/webview/MainActivity.kt
package com.example.webview

import android.graphics.Bitmap
import android.os.Bundle
import android.view.View
import android.webkit.WebView
import android.webkit.WebViewClient
import android.widget.ProgressBar
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    var webView: WebView? = null
    var pgBar: ProgressBar? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        webView = findViewById(R.id.webView)
        pgBar = findViewById(R.id.pgBar)
        webView?.loadUrl("https://www.google.com")
        webView?.setWebViewClient(object : WebViewClient() {
            override fun onPageStarted(view: WebView, url: String, favicon: Bitmap) {
                pgBar?.setVisibility(View.VISIBLE)
                super.onPageStarted(view, url, favicon)
            }

            override fun onPageFinished(view: WebView, url: String) {
                pgBar?.setVisibility(View.GONE)
                super.onPageFinished(view, url)
            }
        })
    }

    override fun onBackPressed() {
        if (webView!!.canGoBack()) {
            webView!!.goBack()
        } else {
            super.onBackPressed()
        }
    }
}
